test/unit/setting_test.rb

```diff
@@ -209,6 +209,20 @@ def test_boolean_true_returns_from_cache
     check_value_returns_from_cache_with :name => "test_cache", :default => true, :value => true, :description => "test foo"
   end
 
+  test "create a setting with collection of possibles values" do
+    options = Setting.set("test_attr", "some_description", "default_value", "full_name", "my_value") {{ :a => "a", :b => "b" }}
+    setting = Setting.create(options)
+    assert_equal setting.send("#{setting.name}_list"), { :a => "a", :b => "b" }
+  end
+
+  test "create a setting with collection of dynamic values" do
+    dynamic_hash = {:a => "a"}
+    options = Setting.set("test_attr", "some_description", "default_value", "full_name", "my_value") { dynamic_hash }
+    setting = Setting.create(options)
+    dynamic_hash[:b] = "b"
+    assert_equal setting.send("#{setting.name}_list"), dynamic_hash
+  end
+
```

Comment:
How about a test that checks an actual block that calculates and returns a hash? (something like the example you gave in the PR description)